deployments/multiple-ragapps/docker-compose.yml

```diff
@@ -6,17 +6,76 @@ services:
     labels:
       - traefik.enable=true
       - traefik.docker.network=ragapp-network
-      # Default username: admin, password: admin
-      - traefik.http.middlewares.admin-auth.basicauth.users=${USERNAME:-admin}:${HASHED_PASSWORD:-$$apr1$$3gG256rA$$BomHrC/uMhL807k/t7Wd6/}
-    volumes:
+      # Keycloak OpenID plugin configuration
+      - traefik.http.middlewares.ragapp-keycloakopenid.plugin.keycloakopenid.KeycloakURL=http://localhost/auth
+      - traefik.http.middlewares.ragapp-keycloakopenid.plugin.keycloakopenid.ClientID=ragapp
+      # Secret from the example ragapp keycloak realm
+      - traefik.http.middlewares.ragapp-keycloakopenid.plugin.keycloakopenid.ClientSecret=jh4yDlJ5QsBSafVNaqqLxEG3vKcYyNgD 
+      - traefik.http.middlewares.ragapp-keycloakopenid.plugin.keycloakopenid.KeycloakRealm=ragapp
+      - traefik.http.middlewares.ragapp-keycloakopenid.plugin.keycloakopenid.Scope=openid
+      # Admin authentication middleware
+      # 1. Use Keycloak user with role admin-manager
+      # Set UserClaimName and UserHeaderName to extract roles from the token and set them in the header for authorization
+      - traefik.http.middlewares.ragapp-keycloakopenid.plugin.keycloakopenid.UserClaimName=roles
+      - traefik.http.middlewares.ragapp-keycloakopenid.plugin.keycloakopenid.UserHeaderName=X-Forwarded-Roles
+      - traefik.http.middlewares.admin-auth.plugin.headauth.header.name=X-Forwarded-Roles
+      - traefik.http.middlewares.admin-auth.plugin.headauth.allowed[0]=[admin-manager]
+      - traefik.http.middlewares.admin-auth.plugin.headauth.methods[0]=GET
+      - traefik.http.middlewares.admin-auth.plugin.headauth.methods[1]=POST
+      - traefik.http.middlewares.admin-auth.plugin.headauth.methods[2]=PUT
+      - traefik.http.middlewares.admin-auth.plugin.headauth.methods[3]=DELETE
+      # 2. Alternative, we can just basicAuth for admin access. Please comment out the Keycloak middleware above and uncomment the following lines
+      # username: admin, password: admin
+      # - traefik.http.middlewares.admin-auth.basicauth.users=${USERNAME:-admin}:${HASHED_PASSWORD:-$$apr1$$3gG256rA$$BomHrC/uMhL807k/t7Wd6/}
```

Comment:
maybe better do a new deployment?

Reply:
It's not a big issue for me because the new deployment will be the same as the current one, except for this middleware.